docs/examples/preprocess.py

```diff
@@ -31,36 +82,107 @@ def preprocess():
+        spaces_at_start_of_line = (len(line) - len(line.lstrip(" "))) * " "
+        reference = line.strip()[len("* :doc:`") :].split("`")[0]
+        logger.debug(f"Example {relative_readme_template_path} has link to {reference}.")
+        referenced_file = (
+            docs_root / reference[1:]
+            if reference.startswith("/examples")
+            else relative_readme_template_path.parent / reference[1:]
+        )
+        referenced_file = referenced_file.relative_to(docs_root)
+        referenced_folder = referenced_file.parent
+        github_link = (
+            f"https://github.com/mila-iqia/mila-docs/tree/master/docs/{referenced_folder}"
+        )
+        new_line = spaces_at_start_of_line + f"* `{referenced_folder} <{github_link}>`_"
+        result[i] = new_line
+        # line = ""
+    return result
+
+
+def inline_docs_for_github_viewing(example_readme_template_path: Path) -> str:
+    relative_readme_template_path = example_readme_template_path.relative_to(docs_root)
+
+    content = example_readme_template_path.read_text()
+    content_lines = content.splitlines()
+    content_lines = (
+        textwrap.dedent(
+            f"""\
+            .. NOTE: This file is auto-generated from {relative_readme_template_path}
+            .. This is done so this file can be easily viewed from the GitHub UI.
+            .. **DO NOT EDIT**
+            """
+        ).splitlines()
+        + [""]  # Need an empty line between this header and the content, otherwise rstcheck cries
+        + content_lines
+    )
+
+    i = 0
+    end = len(content_lines)
+    while i < end:
+        line = content_lines[i]
+        if line.startswith(".. literalinclude:: "):
+            path = line[len(".. literalinclude:: ") :].strip(" ")
+            # TODO: This causes an issue when building the docs because the path to job.sh
+            if path.startswith("examples/"):
+                # all good, the path is absolute already.
+                # logger.debug(f"The path to source code is already absolute: {path}")
+                pass
+            elif path.count("/") == 0:
+                # logger.debug(f"The path to source code is relative: {path}")
+                path = str(example_readme_template_path.parent / path)
             else:
-                i += 1
-        _f.write_text("\n".join(content).replace("\t", " " * 4))
+                raise NotImplementedError(
+                    f"Example {example_readme_template_path} uses a weird format in a "
+                    f"literalinclude: {path}"
+                )
+            lang = ""
+            j = 0
+            for j, line in enumerate(content_lines[i + 1 :]):
+                line = line.strip(" ")
+                if line.startswith(":language:"):
+                    lang = line[len(":language:") :].strip(" ")
+                elif line.startswith(".. literalinclude:: ") or not line:
+                    break
+            del content_lines[i : i + 1 + j]
```

Comment:
Another strategy to remove the literalinclude would be to change `.. literalinclude: ...` to `..`, which would be interpreted as a comment block. If you then put the insert on top, you don't actually need to know where the block ends.

Reply:
Interesting idea. I didn't go for that approach, but I did simplify the code quite a bit in 
34cc66ec8cb48041ee495816412a124f5ea17a8f :

Now it works essentially like this:

```python
old_content_lines = old_content.splitlines() 
new_content: list[str] = [".. This file was generated blablabla", ""]

current_line_index = 0
while line_index < len(old_content_lines)
     line = old_content_lines[line_index]
     if not is_literalinclude_block(line):
         new_content.append(line)
         current_line_index += 1
         continue
     language_line = old_content_lines[line_index+1]
     path: Path = _get_path(line)
     language: str = _get_language(language_line)
     new_content += path.read_text().splitlines()
    current_line_index += 2
```

Reply:
Yeah, that works, although we need to make sure the literalinclude statements are exactly two lines long.

Reply:
Fixed in 83fcb30b5673956fc791f5af08ea2d6d35252943